src/autolabel/data_loaders/validation.py

```diff
@@ -0,0 +1,259 @@
+                    raise ValueError(
+                        f"labels: {unmatched_label} do not match promt/labels provided in config "
+                    )
+            except JSONDecodeError:
+                raise
+        else:
+            raise
+
+
+@dataclass
+class ClassificationTaskValidate:
+    """Validate Classification Task
+
+    The label column can either be a string or a string of list
+    """
+
+    label_column: str
+    labels_set: set  # A classification Task should have a unique set of labels in config
+
+    def validate(self, value: str):
+        """Validate classification
+
+        A classification label(ground_truth) could either be a list or string
+        """
+        # TODO: This can be made better
+        if value.startswith("[") and value.endswith("]"):
+            try:
+                seed_labels = eval(value)
+                if not isinstance(seed_labels, list):
+                    raise
+                unmatched_label = set(seed_labels) - self.labels_set
+                if len(unmatched_label) != 0:
+                    raise ValueError(
+                        f"labels: {unmatched_label} do not match promt/labels provided in config "
+                    )
+            except SyntaxError:
+                raise
+        else:
+            if value not in self.labels_set:
+                raise ValueError(
+                    f"labels: {value} do not match promt/labels provided in config "
+                )
+
+
+@dataclass
+class EMTaskValidate:
+    """Validate Entity Matching Task
+
+    As of now we assume that the input label_column is a string
+    """
+
+    label_column: str
+    labels_set: set  # An EntityMatching Task should have a unique set of labels in config
+
+    def validate(self, value: str):
+        if value not in self.labels_set:
+            raise ValueError(
+                f"labels: {value} do not match promt/labels provided in config "
+            )
+
+
+@dataclass
+class QATaskValidate:
+    """Validate Question Answering Task
+
+    As of now we assume that the input label_column is a string
+    """
+
+    label_column: str
+    labels_set: Optional[
+        set
+    ]  # A QA task may or may not have a unique set of label list
+
+
+TaskTypeValidate = Union[
+    NERTaskValidate,
+    ClassificationTaskValidate,
+    EMTaskValidate,
+    QATaskValidate,
+]
+
+
+@dataclass
+class DataValidationTasks:
+    classification: TaskTypeValidate = ClassificationTaskValidate
+    named_entity_recognition: TaskTypeValidate = NERTaskValidate
+    entity_matching: TaskTypeValidate = EMTaskValidate
+    question_answering: TaskTypeValidate = QATaskValidate
+
+
+def fetch_expected_columns(
+    example_template: str, pattern: str = r"\{([^}]*)\}"
+) -> List:
+    """Fetch desired columns fro example template
+
+    Args:
+        example_template (str): example template from config file
+        pattern (str, optional): regex pattern. Defaults to r"\{([^}]*)}".
```

Comment:
we can just make this the pattern always - don't need to make it a parameter?

Reply:
That works too. Having it as a parameter though would provide an option.

IMO we could have the pattern defined in some file that contains the global variables. that way it can be changed if needed. Thoughts?

Reply:
sure, we can define it at the top of this file for now